common/src/KokkosFFT_traits.hpp

```diff
@@ -261,6 +261,41 @@ struct complex_view_type {
   using type = Kokkos::View<complex_type*, array_layout_type, ExecutionSpace>;
 };
 
+template <typename ExecutionSpace, typename Enable = void>
+struct is_AnyHostSpace : std::false_type {};
+
+template <typename ExecutionSpace>
+struct is_AnyHostSpace<ExecutionSpace,
+                       std::enable_if_t<Kokkos::SpaceAccessibility<
+                           ExecutionSpace, Kokkos::HostSpace>::accessible>>
+    : std::true_type {};
+
+/// \brief Helper to check if the ExecutionSpace is one of the enabled
+/// HostExecutionSpaces
+template <typename ExecutionSpace>
+inline constexpr bool is_AnyHostSpace_v =
+    is_AnyHostSpace<ExecutionSpace>::value;
+
+#if !defined(ENABLE_HOST_AND_DEVICE) &&                           \
+    (defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP) || \
+     defined(KOKKOS_ENABLE_SYCL))
+template <typename ExecutionSpace, typename Enable = void>
+struct is_AllowedSpace : std::false_type {};
+template <typename ExecutionSpace>
+struct is_AllowedSpace<ExecutionSpace,
+                       std::enable_if_t<std::is_same_v<
+                           ExecutionSpace, Kokkos::DefaultExecutionSpace>>>
+    : std::true_type {};
```

Comment:
```suggestion
template <typename ExecutionSpace>
struct is_AllowedSpace : std::is_same<ExecutionSpace, Kokkos::DefaultExecutionSpace> {};
```

I think it should be the same ?